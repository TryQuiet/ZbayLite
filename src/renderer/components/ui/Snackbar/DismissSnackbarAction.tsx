import React from 'react'

import { useSnackbar } from 'notistack'

import Button from '@material-ui/core/Button'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  dismiss: {
    fontWeight: theme.typography.fontWeightMedium,
    color: '#fff'
  }
}))

interface DismissSnackbarActionProps {
  notificationKey: string | number
}

export const DismissSnackbarAction: React.FC<DismissSnackbarActionProps> = ({
  notificationKey
}) => {
  const classes = useStyles({})

  const { closeSnackbar } = useSnackbar()

  return (
    <Button
      onClick={() => {
        closeSnackbar(notificationKey)
      }}
      size='small'
      className={classes.dismiss}>
      Dismiss
    </Button>
  )
}

export const notifierAction = (key: string | number) => <DismissSnackbarAction notificationKey={key} />

export default DismissSnackbarAction
