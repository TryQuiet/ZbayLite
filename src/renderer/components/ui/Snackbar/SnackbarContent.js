import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { withStyles } from '@material-ui/core/styles'
import IconButton from '@material-ui/core/IconButton'
import CircularProgress from '@material-ui/core/CircularProgress'
import SnackbarContentMaterial from '@material-ui/core/SnackbarContent'

import CloseIcon from '@material-ui/icons/Close'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import ErrorIcon from '@material-ui/icons/Error'
import InfoIcon from '@material-ui/icons/Info'
import WarningIcon from '@material-ui/icons/Warning'

import green from '@material-ui/core/colors/green'
import blue from '@material-ui/core/colors/blue'
import amber from '@material-ui/core/colors/amber'

const styles = theme => ({
  close: {
    padding: theme.spacing(0.5),
    margin: 0
  },
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: blue[600]
  },
  loading: {
    backgroundColor: blue[400],
    color: '#fff'
  },
  loadingIcon: {
    color: '#fff',
    opacity: 1
  },
  warning: {
    backgroundColor: amber[700]
  },
  fullWidth: {
    'max-width': 'none',
    'flex-grow': 1
  },
  icon: {
    fontSize: 20,
    opacity: 0.9
  },
  content: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    paddingLeft: theme.spacing(2)
  }
})

const iconVariants = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
  loading: CircularProgress
}

export const SnackbarContent = ({
  classes,
  variant,
  message,
  onClose,
  fullWidth
}) => {
  const Icon = iconVariants[variant]
  const closeAction = (
    <IconButton
      key='close'
      color='inherit'
      className={classes.close}
      onClick={onClose}
    >
      <CloseIcon className={classes.icon} />
    </IconButton>
  )
  const action = variant !== 'loading' ? [closeAction] : []
  return (
    <SnackbarContentMaterial
      className={
        classNames({
          [classes[variant]]: true,
          [classes.fullWidth]: fullWidth
        })
      }
      message={
        <span className={classes.content}>
          <Icon className={classNames(classes.icon, classes[`${variant}Icon`])} size={20} />
          <span className={classes.message}>
            {message}
          </span>
        </span>
      }
      action={action}
    />
  )
}

SnackbarContent.propTypes = {
  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info', 'loading']).isRequired,
  message: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func,
  fullWidth: PropTypes.bool
}

SnackbarContent.defaultProps = {
  fullWidth: false
}

export default React.memo(withStyles(styles)(SnackbarContent))
