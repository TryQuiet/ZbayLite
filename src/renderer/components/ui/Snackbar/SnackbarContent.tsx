import React from 'react'
import classNames from 'classnames'

import { makeStyles } from '@material-ui/core/styles'
import IconButton from '@material-ui/core/IconButton'
import CircularProgress from '@material-ui/core/CircularProgress'
import SnackbarContentMaterial from '@material-ui/core/SnackbarContent'

import CloseIcon from '@material-ui/icons/Close'
import CheckCircleIcon from '@material-ui/icons/CheckCircle'
import ErrorIcon from '@material-ui/icons/Error'
import InfoIcon from '@material-ui/icons/Info'
import WarningIcon from '@material-ui/icons/Warning'

import green from '@material-ui/core/colors/green'
import blue from '@material-ui/core/colors/blue'
import amber from '@material-ui/core/colors/amber'

const useStyles = makeStyles(theme => ({
  close: {
    padding: theme.spacing(0.5),
    margin: 0
  },
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: blue[600]
  },
  loading: {
    backgroundColor: blue[400],
    color: '#fff'
  },
  loadingIcon: {
    color: '#fff',
    opacity: 1
  },
  warning: {
    backgroundColor: amber[700]
  },
  fullWidth: {
    'max-width': 'none',
    'flex-grow': 1
  },
  icon: {
    fontSize: 20,
    opacity: 0.9
  },
  content: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    paddingLeft: theme.spacing(2)
  }
}))

const iconVariants = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
  loading: CircularProgress
}

interface SnackbarContentProps {
  message: string
  variant: 'success' | 'warning' | 'error' | 'info' | 'loading'
  onClose?: () => void
  fullWidth?: boolean
}

export const SnackbarContent: React.FC<SnackbarContentProps> = ({
  message,
  variant,
  onClose,
  fullWidth = false
}) => {
  const classes = useStyles({})

  const Icon = iconVariants[variant]

  const closeAction = (
    <IconButton key='close' color='inherit' className={classes.close} onClick={onClose}>
      <CloseIcon className={classes.icon} />
    </IconButton>
  )

  const action = variant !== 'loading' ? [closeAction] : []
  const variantIcon = `${variant}Icon` as 'close' | 'content' | 'icon' | 'success' | 'error' | 'info' | 'loading' | 'loadingIcon' | 'warning' | 'fullWidth' | 'message'
  return (
    <SnackbarContentMaterial
      className={classNames({
        [classes[variant]]: true,
        [classes.fullWidth]: fullWidth
      })}
      message={
        <span className={classes.content}>
          <Icon className={classNames(classes.icon, classes[variantIcon])} size={20} />
          <span className={classes.message}>{message}</span>
        </span>
      }
      action={action}
    />
  )
}

export default SnackbarContent
