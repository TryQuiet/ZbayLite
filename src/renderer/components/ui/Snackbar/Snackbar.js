import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

import { withStyles } from '@material-ui/core/styles'
import MaterialSnackbar from '@material-ui/core/Snackbar'
import SnackbarContent from './SnackbarContent'

const styles = theme => ({
  fullWidthBottom: {
    left: 0,
    right: 0,
    bottom: 0
  },
  fullWidthTop: {
    left: 0,
    right: 0,
    bottom: 0
  }
})

export const Snackbar = ({
  classes,
  variant,
  open,
  onClose,
  position,
  fullWidth,
  message
}) => (
  <MaterialSnackbar
    anchorOrigin={position}
    open={open}
    autoHideDuration={6000}
    onClose={onClose}
    classes={{
      anchorOriginTopCenter: classNames({
        [classes.fullWidthTop]: fullWidth
      }),
      anchorOriginBottomCenter: classNames({
        [classes.fullWidthBottom]: fullWidth
      }),
      anchorOriginTopRight: classNames({
        [classes.fullWidthTop]: fullWidth
      }),
      anchorOriginBottomRight: classNames({
        [classes.fullWidthBottom]: fullWidth
      }),
      anchorOriginTopLeft: classNames({
        [classes.fullWidthTop]: fullWidth
      }),
      anchorOriginBottomLeft: classNames({
        [classes.fullWidthBottom]: fullWidth
      })
    }}
  >
    <SnackbarContent
      variant={variant}
      message={message}
      fullWidth={fullWidth}
      onClose={onClose}
    />
  </MaterialSnackbar>
)

Snackbar.propTypes = {
  classes: PropTypes.object.isRequired,
  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info', 'loading']).isRequired,
  open: PropTypes.bool.isRequired,
  position: PropTypes.exact({
    vertical: PropTypes.oneOf(['bottom', 'top']),
    horizontal: PropTypes.oneOf(['left', 'right'])
  }).isRequired,
  fullWidth: PropTypes.bool,
  onClose: PropTypes.func,
  message: PropTypes.string.isRequired
}

Snackbar.defaultProps = {
  position: { vertical: 'bottom', horizontal: 'left' },
  fullWidth: false
}

export default React.memo(withStyles(styles)(Snackbar))
