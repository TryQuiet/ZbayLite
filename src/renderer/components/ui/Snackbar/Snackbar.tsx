import React from 'react'

import classNames from 'classnames'

import { makeStyles } from '@material-ui/core/styles'
import MaterialSnackbar from '@material-ui/core/Snackbar'

import SnackbarContent from './SnackbarContent'

const useStyles = makeStyles(() => ({
  fullWidthBottom: {
    left: 0,
    right: 0,
    bottom: 0
  },
  fullWidthTop: {
    left: 0,
    right: 0,
    bottom: 0
  }
}))

interface SnackbarProps {
  open: boolean
  message: string
  variant: 'success' | 'warning' | 'error' | 'info' | 'loading'
  position?: {
    vertical: 'bottom' | 'top'
    horizontal: 'left' | 'right'
  }
  fullWidth?: boolean
  onClose?: () => void
}

export const Snackbar: React.FC<SnackbarProps> = ({
  open,
  message,
  variant,
  onClose,
  position = { vertical: 'bottom', horizontal: 'left' },
  fullWidth = false
}) => {
  const classes = useStyles({})
  return (
    <MaterialSnackbar
      anchorOrigin={position}
      open={open}
      autoHideDuration={6000}
      onClose={onClose}
      classes={{
        anchorOriginTopCenter: classNames({
          [classes.fullWidthTop]: fullWidth
        }),
        anchorOriginBottomCenter: classNames({
          [classes.fullWidthBottom]: fullWidth
        }),
        anchorOriginTopRight: classNames({
          [classes.fullWidthTop]: fullWidth
        }),
        anchorOriginBottomRight: classNames({
          [classes.fullWidthBottom]: fullWidth
        }),
        anchorOriginTopLeft: classNames({
          [classes.fullWidthTop]: fullWidth
        }),
        anchorOriginBottomLeft: classNames({
          [classes.fullWidthBottom]: fullWidth
        })
      }}>
      <SnackbarContent
        variant={variant}
        message={message}
        fullWidth={fullWidth}
        onClose={onClose}
      />
    </MaterialSnackbar>
  )
}

export default Snackbar
