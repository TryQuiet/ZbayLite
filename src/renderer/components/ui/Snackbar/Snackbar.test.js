import React from 'react'

import { shallow } from 'enzyme'

import { Snackbar } from './Snackbar'

import each from 'jest-each'

describe('Snackbar', () => {
  each(['success', 'warning', 'error', 'info', 'loading']).test('renders %s', variant => {
    const result = shallow(
      <Snackbar open message='test snackbar' variant={variant} onClose={jest.fn()} />
    )
    expect(result).toMatchSnapshot()
  })

  each([
    ['top', 'left'],
    ['top', 'right'],
    ['bottom', 'left'],
    ['top', 'right']
  ]).test('renders full width for position [%s, %s]', (vertical, horizontal) => {
    const result = shallow(
      <Snackbar
        open
        message='test snackbar'
        variant='success'
        position={{ vertical, horizontal }}
        fullWidth
        onClose={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('renders closed', () => {
    const result = shallow(
      <Snackbar open={false} message='test snackbar' variant='success' onClose={jest.fn()} />
    )
    expect(result).toMatchSnapshot()
  })
})
