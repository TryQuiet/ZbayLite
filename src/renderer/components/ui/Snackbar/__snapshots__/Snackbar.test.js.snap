// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar renders closed 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={false}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders error 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="error"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders full width for position [bottom, left] 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "fullWidthBottom-mock",
      "anchorOriginBottomLeft": "fullWidthBottom-mock",
      "anchorOriginBottomRight": "fullWidthBottom-mock",
      "anchorOriginTopCenter": "fullWidthTop-mock",
      "anchorOriginTopLeft": "fullWidthTop-mock",
      "anchorOriginTopRight": "fullWidthTop-mock",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={true}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders full width for position [top, left] 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "fullWidthBottom-mock",
      "anchorOriginBottomLeft": "fullWidthBottom-mock",
      "anchorOriginBottomRight": "fullWidthBottom-mock",
      "anchorOriginTopCenter": "fullWidthTop-mock",
      "anchorOriginTopLeft": "fullWidthTop-mock",
      "anchorOriginTopRight": "fullWidthTop-mock",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={true}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders full width for position [top, right] 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "fullWidthBottom-mock",
      "anchorOriginBottomLeft": "fullWidthBottom-mock",
      "anchorOriginBottomRight": "fullWidthBottom-mock",
      "anchorOriginTopCenter": "fullWidthTop-mock",
      "anchorOriginTopLeft": "fullWidthTop-mock",
      "anchorOriginTopRight": "fullWidthTop-mock",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={true}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders full width for position [top, right] 2`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "fullWidthBottom-mock",
      "anchorOriginBottomLeft": "fullWidthBottom-mock",
      "anchorOriginBottomRight": "fullWidthBottom-mock",
      "anchorOriginTopCenter": "fullWidthTop-mock",
      "anchorOriginTopLeft": "fullWidthTop-mock",
      "anchorOriginTopRight": "fullWidthTop-mock",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={true}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders info 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="info"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders loading 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="loading"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders success 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="success"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`Snackbar renders warning 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  classes={
    Object {
      "anchorOriginBottomCenter": "",
      "anchorOriginBottomLeft": "",
      "anchorOriginBottomRight": "",
      "anchorOriginTopCenter": "",
      "anchorOriginTopLeft": "",
      "anchorOriginTopRight": "",
    }
  }
  onClose={[MockFunction]}
  open={true}
>
  <Memo(WithStyles(SnackbarContent))
    fullWidth={false}
    message="test snackbar"
    onClose={[MockFunction]}
    variant="warning"
  />
</WithStyles(ForwardRef(Snackbar))>
`;
